pkg/container-utils/cri/cri.go

```diff
@@ -82,16 +82,61 @@ func listContainers(c *CRIClient, filter *runtime.ContainerFilter) ([]*runtime.C
 	return res.GetContainers(), nil
 }
 
+func listPodSandboxes(c *CRIClient, filter *runtime.PodSandboxFilter) ([]*runtime.PodSandbox, error) {
+	podRequest := &runtime.ListPodSandboxRequest{}
+	if filter != nil {
+		podRequest.Filter = filter
+	}
+
+	podRes, err := c.client.ListPodSandbox(context.Background(), podRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	return podRes.Items, nil
+}
+
+func getPodSandbox(c *CRIClient, podSandboxID string) (*runtime.PodSandbox, error) {
+	podSandboxes, err := listPodSandboxes(c, &runtime.PodSandboxFilter{
+		Id: podSandboxID,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(podSandboxes) == 0 {
+		return nil, fmt.Errorf("pod sandbox %q not found", podSandboxID)
+	}
+	if len(podSandboxes) > 1 {
+		log.Warnf("CRIClient: found multiple pod sandboxes (%d) with ID %q. Taking the first one: %+v",
+			len(podSandboxID), podSandboxID, podSandboxes)
```

Comment:
Imo it should be a fatal error. That should never happen.
I looked at `func (c *CRIClient) GetContainer` and thought there was a reason behind it.
Since you also think this is weird, I'll change it also for `GetContainer`